### DISCUSSION TITLE
Inverse Kinematics formula derivation from MoveIt?

### DISCUSSION THREAD

**Discussion by Ivoferrer9** [moveit/moveit2#2510] [Q&A]
Is there a way for me to pullout the inverse kinematics formula derived from Kdl solver on my URDF?  I am just a beginner with ros and I don't really know how to get the movements I get from rviz and push that to my arduino. Is it possible to just pull out the inverse kinematics formula and just assign values to it on the arduino?

**Comment by bmaxdk** [comment]
The KDL solver does not produce a static, one-size-fits-all formula that you can easily extract and use elsewhere (like on an Arduino). Instead, it dynamically computes the IK solutions based on the current state and configuration of the robot as defined in the URDF.

If you're looking to use the results of the KDL solver on Arduino, you'd have to compute the IK solutions in ROS and then send these computed joint angles or positions to the Arduino via a communication protocol. Directly extracting a formula for use on the Arduino is not feasible.
